vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -268,118 +676,90 @@ class StartNode {
     size_t index_;
 };
 
-// Minimum size of search for channels to profile.  kMinProfile results
-// in searching x = [0, kMinProfile], and y = [0, kMinProfile[.
-//
-// Making this value larger will increase the sample size, but also the runtime
-// to produce the lookahead.
-static constexpr int kMinProfile = 1;
-
-// Maximum size of search for channels to profile.  Once search is outside of
-// kMinProfile distance, lookahead will stop searching once:
-//  - At least one channel has been profiled
-//  - kMaxProfile is exceeded.
-static constexpr int kMaxProfile = 7;
+static void compute_connection_box_lookahead(
+    const std::vector<t_segment_inf>& segment_inf,
+    const std::string& /*search_locations_str*/) {
+    vtr::ScopedStartFinishTimer timer("Computing connection box lookahead map");
 
-static int search_at(int iseg, int start_x, int start_y, t_routing_cost_map* cost_map) {
-    const auto& device_ctx = g_vpr_ctx.device();
+    auto& device_ctx = g_vpr_ctx.device();
 
-    int count = 0;
-    int dx = 0;
-    int dy = 0;
+    // First find bounding boxes for all segments.
+    std::vector<t_bb> segment_bb(segment_inf.size(), t_bb{INT_MAX, INT_MIN, INT_MAX, INT_MIN});
```

Comment:
`INT_MAX` and `INT_MIN` require the `<climits>` lib. 
I also think that a better way would be to use `std::numeric_limits<int>::max()` and include `<limits>`

Reply:
Agreed